sycl/source/detail/non_cached_kernel_lock.cpp

```diff
@@ -0,0 +1,22 @@
+#include "non_cached_kernel_lock.hpp"
+
+__SYCL_INLINE_NAMESPACE(cl) {
+namespace sycl {
+namespace detail {
+
+NonCachedKernelLock::~NonCachedKernelLock() {}
+
+Locked<RT::PiKernel> NonCachedKernelLock::lockKernel(RT::PiKernel &K) {
```

Comment:
the same, `RT::PiKernel` is just `void*`, so `&` is not required here.

Reply:
This reference here is to satisfy `Locked` which holds reference.
Otherwise, I'd have to introduce `LockedPtr` which holds pointer.